OPHD/MicroPather/micropather.cpp

```diff
@@ -850,7 +850,7 @@ void MicroPather::GetCacheData(CacheData* data)
 		data->miss = pathCache->miss;
 		if (data->hit + data->miss)
 		{
-			data->hitFraction = static_cast<float>(static_cast<double>(data->hit) / static_cast<double>(data->hit + data->miss));
+			data->hitFraction = static_cast<float>(static_cast<double>(data->hit) / (static_cast<double>(data->hit) + data->miss));

```

Comment:
This doesn't appear to make a whole lot of sense. Both `hit` and `miss` are declared as `int`. It seems odd to explicitly cast one of them to `double`, and then rely on the other to be implicitly converted to `double`.

Additionally, the expression `data->hit + data->miss` is re-used in the `if` condition. Maybe we should create an `auto total` temporary variable to hold the result. I believe casting the temporary to `double` would eliminate the warning, and perhaps make more sense. Granted, that is still doing what the warning was about, naming doing calculations on the narrower size, and then casting.

The other thought is the `if` condition is checking for a condition that would lead to divide by 0. We should perhaps be using the same type for the `if` check as is actually used in the calculation, otherwise we still risk division by 0 due to conversion issues.

It might also make more sense to just use `float` throughout, rather than cast between `double` and `float`.